src/renderer/components/editorWithTabs/tabs.vue

```diff
@@ -18,33 +27,64 @@
             <use id="default-close-icon" xlink:href="#icon-close-small"></use>
           </svg>
         </li>
-        <li class="new-file">
-          <svg class="icon" aria-hidden="true"
-            @click.stop="newFile()"
-          >
-            <use xlink:href="#icon-plus"></use>
-          </svg>
-        </li>
       </ul>
     </div>
+    <div
+      class="new-file"
+    >
+      <svg class="icon" aria-hidden="true"
+        @click.stop="newFile()"
+      >
+        <use xlink:href="#icon-plus"></use>
+      </svg>
+    </div>
+  </div>
 </template>
 
 <script>
   import { mapState } from 'vuex'
+  import dragula from 'dragula'
   import { tabsMixins } from '../../mixins'
 
   export default {
+    data () {
+      return {}
+    },
     mixins: [tabsMixins],
     computed: {
       ...mapState({
-        currentFile: state => state.editor.currentFile,
-        tabs: state => state.editor.tabs
+        'currentFile': state => state.editor.currentFile,
+        'tabs': state => state.editor.tabs,
+        'showSideBar': state => state.layout.showSideBar,
+        'sideBarWidth': state => state.layout.sideBarWidth
       })
     },
     methods: {
       newFile () {
         this.$store.dispatch('NEW_BLANK_FILE')
+      },
+      handleTabScroll (event) {
+        const tabs = this.$refs.tabContainer
+        const newLeft = Math.max(0, Math.min(tabs.scrollLeft + event.deltaY, tabs.scrollWidth ))
+        tabs.scrollLeft = newLeft
       }
+    },
+    mounted () {
+      this.$nextTick(() => {
+        const tabs = this.$refs.tabContainer
+        tabs.addEventListener('wheel', this.handleTabScroll)
+
+        dragula([ this.$refs.tabDropContainer ], {
+          direction: 'horizontal',
+          revertOnSpill: true,
+          mirrorContainer: this.$refs.tabDropContainer,
+          ignoreInputTextSelection: false
+        })
+      })
+    },
+    beforeDestroy () {
+      const tabs = this.$refs.tabContainer
```

Comment:
Does `dragula` has a destroy-like method, so that we can destroy the instance of dragula?

FYI, I found this:

>drake.destroy()
>Removes all drag and drop events used by dragula to manage drag and drop between the containers. If >.destroy is called while an element is being dragged, the drag will be effectively cancelled.